### ISSUE TITLE
Article "Time​Interval, Date, and Date​Interval" - `NSDate`

### ISSUE THREAD

**Issue by helje5** [NSHipster/articles#532]
> Why NSDate and not NSTime? Our guess is that the originators of this API wanted to match its counterpart in java.util.date when EOF targeted both Java and Objective-C.

EOF (and hence Foundation) predates the Java efforts of NeXT significantly. WO 5.0 (the Java version) was only released ~2001 (and Java 1.0 itself in 1996 only). I don't think this had any relevance here. I would suspect that the reverse might be true (the Java engineer borrowing the name from Foundation, remember that before Java [OpenStep for Solaris](http://www.kevra.org/TheBestOfNext/NeXTProducts/NeXTSoftware/OpenStep-SolarisWhitePaper/files/page538_1.pdf) was a thing!).

Why not `NSTime`? Well, `NSTime` would be wrong too (time is something like 3:40pm). I think technically a better name could have been `NSTimestamp`.

Also remember that a lot of that happened in a time when SQL databases have been a major thing in business environments. Those already supported "large dates" (funny enough also the separate concepts of `date`, `time` and `datetime`, including the `with timezone` variants).

I think it is worth mentioning that originally there was not just `NSDate` but also [`NSCalendarDate`](https://developer.apple.com/documentation/foundation/nscalendardate), a subclass of `NSDate`. An `NSCalendarDate` coupled the `NSDate` "timestamp" w/ a timezone, thereby giving it the context to perform calendrical operations.
Though that API had issues (coupling the two difficult concepts too tightly), that kind of object is actually very useful. Many times you actually want to store and keep stuff as "datetime with timezone", not as a plain UTC timestamp. ([also floating](http://www.alwaysrightinstitute.com/time-is-crazy/)  etc)

Wrt the reference date being "a hedge against Y2K", I think this is very likely true as that started to be a topic back then. 

In summary: Dates and times and timezones are really hard. Especially to get them right from the beginning 😉
